Makefile

```diff
@@ -120,51 +142,68 @@ DEPENDS	:=	$(OFILES:.o=.d)
 #---------------------------------------------------------------------------------
 # main targets
 #---------------------------------------------------------------------------------
-all: $(OUTPUT).gcb $(OUTPUT).vgc $(OUTPUT).gci $(OUTPUT)_xz.gci
 
+#---------------------------------------------------------------------------------
+# Base DOL (PicoBoot)
+#---------------------------------------------------------------------------------
 $(OUTPUT).dol: $(OUTPUT).elf
 $(OUTPUT).elf: $(OFILES)
 
 $(OFILES_SOURCES) : $(HFILES)
 
-%.gcb: %.dol
-	@echo pack IPL ... $(notdir $@)
-	@cd $(PWD); ./dol2ipl.py ipl.rom $< $@
-
+#---------------------------------------------------------------------------------
+# GCI
+#---------------------------------------------------------------------------------
 %.gci: %.dol
-	@echo dol2gci ... $(notdir $@)
+	@echo converting DOL to GCI ... $(notdir $@)
 	@dol2gci $< $@ boot.dol
 
-$(OUTPUT)_xz.dol: $(OUTPUT).dol
-	@echo compress ... $(notdir $@)
-	@dolxz $< $@ -cube
-
-$(OUTPUT)_xz.elf: $(OUTPUT)_xz.dol
-	@echo dol2elf ... $(notdir $@)
-	@doltool -e $<
+#---------------------------------------------------------------------------------
+# Qoob
+#---------------------------------------------------------------------------------
+$(OUTPUT).gcb: $(OUTPUT).dol
```

Comment:
Previously a generic rule, ` %.gcb: %.dol`. I see multiple instances of this. Why the change?

Reply:
I was purposely shifting from a generic to specific target mentality. As in, this is not a target for building a `.gcb` file, this is **the** target for building **the** `iplbool.gcb` Qoob IPL file. It's functionally the same but I think it's a lot easier to read and follow the dependency chain, especially when you get a bunch of overlapping/similar targets such as: `%.elf`, `%_xeno.elf`, `$(OUTPUT).elf`, `$(OUTPUT)_xz.elf`, `$(OUTPUT_SX).elf`. I also don't see the advantage of being abstract.

To me, `%.gci: %.dol` makes sense as we really are converting any ol' DOL into a GCI with no context.

Reply:
Yet you could pass just about any DOL here, and it would boot just fine from a Qoob Pro. The only real restriction is that it can't rely on any hardware being initialized beforehand. For most homebrew that's fine, games rely on the IPL doing some initialization but we don't care about those.

Reply:
Right. But for this project our goal is to build **the** IPLBoot Qoob flash file from **the** IPLBoot DOL. For example, I'd rather `make fubar.gcb` throw "no such target" rather than a cryptic build error about `fubar.elf` failing to link (because `%.gcb -> %.dol -> %.elf` which does not exist).

As you point out, since there is currently nothing about the DOL, GCB, or its conversion that is specific to IPLBoot so it's all functionally the same. My argument is purely conceptional. Perhaps I am being pedantic. :)

Reply:
> especially when you get a bunch of overlapping/similar targets such as: `%.elf`, `%_xeno.elf`, `$(OUTPUT).elf`, `$(OUTPUT)_xz.elf`, `$(OUTPUT_SX).elf`

These are the way they are because that was the only way I could make them work. Make is a very annoying tool to work with.
Either way, I think you're right. Let's go ahead with this change then.